### PR TITLE
juju/txn with server side transactions

### PR DESCRIPTION
**Pull request by jameinel**
This adds a boolean to the NewRunnerParams that allows us to request that we used server-side transactions instead of normal client-side transactions.

It does a check to see if the Mongo database supports server side transactions and gives a log message if it was requested but wasn't available. Would it be better to have that at Warning level, since a user requested it?

### CODE REVIEW

**Comment by mitechie** [comment]
It also seems that would be an error if the user is trying to use SST and
the db doesn't support it.

File: txn.go [hunk 1]
```diff
@@ -186,18 +189,31 @@ type RunnerParams struct {
...
 
 // NewRunner returns a Runner which runs transactions for the database specified in params.
 // Collection names used to manage the transactions and change log may also be specified in
 // params, but if not, default values will be used.
 func NewRunner(params RunnerParams) Runner {
+	sstxn := params.ServerSideTransactions
+	if sstxn {
+		sstxn = SupportsServerSideTransactions(params.Database)
+		if !sstxn {
+			logger.Infof("server-side transactions requested, but database does not support them")
```
Comment (manadart) [inline txn.go]: My instinct is to log it at warning level and also note that client-side transactions will be used.

**Comment by jameinel** [comment]
!!build!!

**Comment by jameinel** [comment]
!!build!!

**Comment by jameinel** [comment]
@mitechie the func would need to be changed to support an error return if we wanted to make it an error, and didn't want to panic() and I definitely wouldn't want to panic.
Also, if someone sets the feature flag for server-side transactions, but they don't ask for mongo4, do we want the API server to just crash and never come up? Do we want bootstrap to fail?
Note that bootstrap into k8s will use mongo 4 by default (without a feature flag), so you'd only have to turn on sstxn, you don't have to also set the 'use-mongo4-from-snap' feature flag.
My thought was that if they set the Mongo4 feature flag, we can try to use them, and just log noisily if it was asked for but we aren't able to actually do the request.
I'm happy with Warning, as that will be as visible as we can make it. I think falling back to Client Side Transactions is still better than just not letting the API server start at all.

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**